### PR TITLE
landing page: change export to download

### LINKED ISSUE #1096: landing page: change data export to download
Instead of displaying the data export on a new page, download a file with the data.

### PR DESCRIPTION
**Pull request by jennur**
closes #1096

### CODE REVIEW

File: invenio_app_rdm/config.py [hunk 1]
```diff
@@ -484,27 +484,37 @@ def files_rest_permission_factory(obj, action):
 APP_RDM_RECORD_EXPORTERS = {
     "json": {
         "name": _("JSON"),
-        "serializer": ("flask_resources.serializers:JSONSerializer")
+        "serializer": ("flask_resources.serializers:JSONSerializer"),
+        "content-type": "application/json",
+        "filename": "{id}.json"
```
Comment (fenekku) [inline invenio_app_rdm/config.py:489]: I understand those were placed here to simplify the task (don't need to touch invenio-rdm-records). Maybe in the future, putting the `content-type` and `filename` in the serializer directly would nicely group coupled responsibility, reduce configuration and keep it dry. It makes sense that the `JSONSerializer` has a content-type of `application/json` and so on. Repeating it is tedious :) . I made an issue for it: https://github.com/inveniosoftware/invenio-rdm-records/issues/817

File: invenio_app_rdm/records_ui/views/records.py [hunk 2]
```diff
@@ -117,15 +117,14 @@ def record_export(
...
-        permissions=record.has_permissions_to(['update_draft']),
-        is_preview=is_preview,
-        is_draft=record._record.is_draft
-    )
+    contentType = exporter.get("content-type", export_format)
+    filename = exporter.get("filename", export_format).format(id=pid_value)
+    headers = {
+        'Content-Type': contentType,
+        'Content-Disposition': 'attachment; filename={filename}'
+        .format(filename=filename),
```
Comment (fenekku) [inline invenio_app_rdm/records_ui/views/records.py:125]: minor:
```suggestion
        'Content-Disposition': f'attachment; filename={filename}',
```

File: invenio_app_rdm/records_ui/views/records.py [hunk 3]
```diff
@@ -145,11 +144,11 @@ def record_file_preview(
...
-            "invenio_app_rdm_records.record_file_download",
-            pid_value=pid_value,
-            filename=file_metadata.data["key"],
-            preview=1 if is_preview else 0
-        )
+        "invenio_app_rdm_records.record_file_download",
+        pid_value=pid_value,
+        filename=file_metadata.data["key"],
+        preview=1 if is_preview else 0
+    )
```
Comment (fenekku) [inline invenio_app_rdm/records_ui/views/records.py:151]: thanks!

**Review by fenekku — APPROVED** [review]
A minor thing and a discussion / new ticket for the future. Thanks!

**Comment by lnielsen** [comment]
Integrated in #1099

**Pull request closed without merging**